src/compiler/execution/DaphneIrExecutor.cpp

```diff
@@ -300,23 +302,34 @@ void DaphneIrExecutor::buildCodegenPipeline(mlir::PassManager &pm) {
     pm.addPass(mlir::daphne::createDaphneOptPass());
 
     if (!userConfig_.use_mlir_hybrid_codegen) {
-        pm.addPass(mlir::daphne::createMatMulOpLoweringPass());
+        pm.addPass(mlir::daphne::createMatMulOpLoweringPass(
+        userConfig_.matmul_tile,
+        userConfig_.matmul_vec_size_bits,
+        userConfig_.matmul_fixed_tile_sizes,
+        userConfig_.matmul_use_fixed_tile_sizes,
+        userConfig_.matmul_unroll_factor));
+        if (userConfig_.explain_mlir_codegen)
+        pm.addPass(
+            mlir::daphne::createPrintIRPass("IR directly after lowering MatMulOp."));
     }
 
-    pm.addPass(mlir::daphne::createAggAllOpLoweringPass());
-    pm.addPass(mlir::daphne::createMapOpLoweringPass());
+    // pm.addPass(mlir::daphne::createAggAllOpLoweringPass());
+    // pm.addPass(mlir::daphne::createMapOpLoweringPass());
     pm.addPass(mlir::createInlinerPass());
 
-    pm.addPass(mlir::daphne::createEwOpLoweringPass());
+    // pm.addPass(mlir::daphne::createEwOpLoweringPass());
     pm.addPass(mlir::createConvertMathToLLVMPass());
-    pm.addPass(mlir::daphne::createModOpLoweringPass());
+    // pm.addPass(mlir::daphne::createModOpLoweringPass());
     pm.addPass(mlir::createCanonicalizerPass());
     pm.addPass(mlir::createCSEPass());
-    pm.addNestedPass<mlir::func::FuncOp>(mlir::createLoopFusionPass());
+    //pm.addNestedPass<mlir::func::FuncOp>(mlir::createLoopFusionPass());
```

Comment:
Please make sure the normal pipeline still runs as expected.

Reply:
I moved some of those passes before the matmul lowering, so that the loop fusion pass doesn't interfere with the loops we create that are potentially intentionally tiled.